docroot/modules/custom/dcc_register_form/src/Form/RegisterForm.php

```diff
@@ -0,0 +1,514 @@
+<?php
+/**
+ * @file
+ * Contains \Drupal\dcc_register_form\Form\RegisterForm
+ */
+
+namespace Drupal\dcc_register_form\Form;
+
+use Drupal\Core\Ajax\AjaxResponse;
+use Drupal\Core\Ajax\PrependCommand;
+use Drupal\Core\Ajax\ReplaceCommand;
+use Drupal\Core\Entity\Entity;
+use Drupal\Core\Entity\EntityTypeManager;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Render\Element\StatusMessages;
+use Drupal\country;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class RegisterForm extends FormBase {
+  protected $entityTypeManager;
+
+  public function __construct(EntityTypeManager $entityTypeManager) {
+    $this->entityTypeManager = $entityTypeManager;
+  }
+
+  public static function create(ContainerInterface $containerInterface) {
+    return new static(
+      $containerInterface->get('entity_type.manager')
+    );
+  }
+
+  public function getFormId() {
+    return 'dcc_register_form';
+  }
+
+  public function buildForm(array $form, FormStateInterface $form_state) {
+
+    $step = $form_state->get("step");
+    if ($step == NULL) {
+      $form_state->set('step', 1);
+      $step = $form_state->get("step");
+    }
+    if ($form_state->getTriggeringElement()['#value'] == 'Next') {
+      if ($step == 1) {
+        $form_state->set("first_name", $form_state->getValue("first_name"));
+        $form_state->set("last_name", $form_state->getValue("last_name"));
+        $form_state->set("email", $form_state->getValue("email"));
+        $form_state->set("drupal_user", $form_state->getValue("drupal_user"));
+        $form_state->set("phone", $form_state->getValue("phone"));
+        $form_state->set("age", $form_state->getValue("age"));
+        $form_state->set("gender", $form_state->getValue("gender"));
+      }
+      if ($step == 2) {
+        $form_state->set("country", $form_state->getValue("country"));
+        $form_state->set("city", $form_state->getValue("city"));
+        $form_state->set("address", $form_state->getValue("address"));
+      }
+      if ($step == 3) {
+        $form_state->set("occupation", $form_state->getValue("occupation"));
+        $form_state->set("organization", $form_state->getValue("organization"));
+        $form_state->set("industry_experience", $form_state->getValue("industry_experience"));
+      }
+
+
+      $form_state->set('step', $form_state->get('step') + 1);
+
+    }
+    if ($form_state->getTriggeringElement()['#value'] == 'Back') {
+      $form_state->set('step', $form_state->get('step') - 1);
+
+    }
+    $form['container'] = array(
+      '#type' => 'container',
+      '#attributes' => array(
+        'id' => 'ajaxcontainer',
+      ),
+    );
+    $step = $form_state->get('step');
+    switch ($step) {
+      case 1:
+        $form['container']['title'] = array(
+          '#title' => "Personal Informations",
+          '#type' => 'item',
+        );
+        if ($form_state->get("first_name")) {
```

Comment:
This if can be made at the default value, with short statement for if. See http://stackoverflow.com/questions/4461996/short-if-else-statement